External/FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -703,151 +687,160 @@ namespace FEXCore::Context {
-          Thread->OpDispatcher->InvalidOp(DecodedInfo);
-          Thread->OpDispatcher->_ExitFunction(Thread->OpDispatcher->_EntrypointOffset(Block.Entry - GuestRIP, GPRSize));
-        }
+          else {
+            // Invalid instruction
+            Thread->OpDispatcher->InvalidOp(DecodedInfo);
+            Thread->OpDispatcher->_ExitFunction(Thread->OpDispatcher->_EntrypointOffset(Block.Entry - GuestRIP, GPRSize));
+          }
+
+          // If we had a dispatch error then leave early
+          if (HadDispatchError) {
+            if (TotalInstructions == 0) {
+              // Couldn't handle any instruction in op dispatcher
+              Thread->OpDispatcher->ResetWorkingList();
+              return { nullptr, nullptr, 0, 0, 0, 0 };
+            }
+            else {
+              const uint8_t GPRSize = GetGPRSize();
 
-        // If we had a dispatch error then leave early
-        if (HadDispatchError) {
-          if (TotalInstructions == 0) {
-            // Couldn't handle any instruction in op dispatcher
-            Thread->OpDispatcher->ResetWorkingList();
-            return { nullptr, nullptr, 0, 0, 0, 0 };
+              // We had some instructions. Early exit
+              Thread->OpDispatcher->_ExitFunction(Thread->OpDispatcher->_EntrypointOffset(Block.Entry + BlockInstructionsLength - GuestRIP, GPRSize));
+              break;
+            }
           }
-          else {
-            const uint8_t GPRSize = GetGPRSize();
 
-            // We had some instructions. Early exit
-            Thread->OpDispatcher->_ExitFunction(Thread->OpDispatcher->_EntrypointOffset(Block.Entry + BlockInstructionsLength - GuestRIP, GPRSize));
+          if (Thread->OpDispatcher->FinishOp(DecodedInfo->PC + DecodedInfo->InstSize, i + 1 == InstsInBlock)) {
             break;
           }
         }
-
-        if (Thread->OpDispatcher->FinishOp(DecodedInfo->PC + DecodedInfo->InstSize, i + 1 == InstsInBlock)) {
-          break;
-        }
       }
+      
+      Thread->OpDispatcher->Finalize();
+
+      Thread->FrontendDecoder->DelayedDisownBuffer();
     }
 
-    Thread->OpDispatcher->Finalize();
+    IR::IREmitter *IREmitter = Thread->OpDispatcher.get();
 
     // Debug
     {
       if (Thread->CTX->Config.DumpIR() != "no") {
-        IRDumper(Thread, GuestRIP, nullptr);
+        IRDumper(Thread, IREmitter, GuestRIP, nullptr);
       }
 
       if (Thread->CTX->Config.ValidateIRarser) {
-        ValidateIR(this, Thread);
+        ValidateIR(this, IREmitter);
       }
     }
 
     // Run the passmanager over the IR from the dispatcher
-    Thread->PassManager->Run(Thread->OpDispatcher.get());
+    Thread->PassManager->Run(IREmitter);
 
     // Debug
     {
       if (Thread->CTX->Config.DumpIR() != "no") {
-        IRDumper(Thread, GuestRIP, Thread->PassManager->HasPass("RA") ? Thread->PassManager->GetPass<IR::RegisterAllocationPass>("RA")->GetAllocationData() : nullptr);
-      }
-
-      if (Thread->OpDispatcher->ShouldDump) {
```

Comment:
Same comment as the other PR here.
What replaces this ShouldDump? It's wildly useful to have around when implementing new instructions.